src/librustc_codegen_llvm/callee.rs

```diff
@@ -37,12 +37,12 @@ pub fn get_fn(
         return llfn;
     }
 
-    let sig = instance.fn_sig(cx.tcx());
     let sym = tcx.symbol_name(instance).name.as_str();
-    debug!("get_fn({:?}: {:?}) => {}", instance, sig, sym);
+    debug!("get_fn({:?}: {:?}) => {}", instance, instance.fn_sig(cx.tcx()), sym);
 
     let llfn = if let Some(llfn) = cx.get_declared_value(&sym) {
         // Create a fn pointer with the substituted signature.
+        let sig = instance.fn_sig(cx.tcx());
         let fn_ptr_ty = tcx.mk_fn_ptr(sig);
         let llptrty = cx.backend_type(cx.layout_of(fn_ptr_ty));
```

Comment:
This is a bad idea, it would erase anything `FnAbi::of_instance` might need to tweak.
So both arms of this if/else should use the same `FnAbi`.
In general, uses of `Instance::fn_sig` (or getting the signature some other way, which may mean any `fn_sig` call anywhere) may require an audit.

Reply:
→ TODO comment, seeing as this is pre-existing.